src/backend/distributed/commands/utility_hook.c

```diff
@@ -826,6 +811,26 @@ CurrentSearchPath(void)
 }
 
 
+/*
+ * PostStandardProcessUtilityChangeGlobalState performs operations to alter
+ * global state of citus utility hook. Those operations should be done
+ * after standard process utility executes even if it errors out.
+ */
+static void
+PostStandardProcessUtilityChangeGlobalState(Node *parsetree)
```

Comment:
What about DecreaseCitusUtilityCounters?

Reply:
It is a better alternative. However, I wanted this method to be a generic one so in future we can put operations -to be done after standard process utility execution" into this function (like I'm planning integrate RevertTruncateTriggerStates logic in citus local tables branch:) )

Maybe I can use the name you proposed and then rename the function later ?

Reply:
I think while making the function generic, you can call DecreaseCitusUtilityCounters within that function.

Reply:
Can we move `InvalidateForeignKeyGraphForDDL ()` here?

Reply:
> Can we move InvalidateForeignKeyGraphForDDL () here?

it works after moving, as expected

Reply:
> I think while making the function generic, you can call DecreaseCitusUtilityCounters within that function.

now that this function does something else plus decrementing global variables, I should find another name I guess :)

Reply:
:D what about something like this:


```c
PostProcessCitusUtility () {
DecreaseCitusUtilityCounters
InvalidateForeignKeyGraphForDDL
}
```

The current name seems a bit too long and generic, feel free to suggest alternatives :)

Reply:
ping https://github.com/citusdata/citus/pull/3990/commits/a4fff2aa2e4309fef3c112c9af2d5918b54142cc